content/intro-to-storybook/angular/en/screen.md

```diff
@@ -321,9 +324,75 @@ Cycling through states in Storybook makes it easy to test we’ve done this corr
   />
 </video>
 
+## Interactive stories
+
+So far, we've been able to build a fully functional application from the ground up, starting from a simple component up to a screen, continuously testing each change using our stories. But with each story added, we would also need to check each variation manually to ensure the UI doesn't break, which requires extra work.
+
+Couldn't we automate this workflow and interact with our components automatically?
+
+Storybook's [`play`](https://storybook.js.org/docs/angular/writing-stories/play-function) function allows us to do just that. They are small snippets of code that run after the story renders. Ideal for scenarios such as ours where we would like to see what happens to our UI when we update our tasks. Since they rely on using framework-agnostic DOM APIs, we can write stories with the play function to interact with the UI and automate human behavior no matter the framework we use.
+
+Let's see it in action! Update your newly created `pure-inbox-screen` story, and set up component interactions by adding the following:
+
+```diff:title=src/app/components/pure-inbox-screen.stories.ts
+import { moduleMetadata, Meta, Story } from '@storybook/angular';
+
++ import { fireEvent, within } from '@storybook/testing-library';
+
+import { CommonModule } from '@angular/common';
+
+import { PureInboxScreenComponent } from './pure-inbox-screen.component';
+import { TaskModule } from './task.module';
+
+import { Store, NgxsModule } from '@ngxs/store';
+import { TasksState } from '../state/task.state';
+
+export default {
+  component:PureInboxScreenComponent,
+  decorators: [
+    moduleMetadata({
+      imports: [CommonModule,TaskModule,NgxsModule.forRoot([TasksState])],
+      providers: [Store],
+    }),
+  ],
+  title: 'PureInboxScreen',
+} as Meta;
+
+const Template: Story = (args) => ({
+  props: args,
+});
+
+export const Default = Template.bind({});
+
+export const Error = Template.bind({});
+Error.args = {
+  error: true,
+};
+
++ export const WithInteractions = Template.bind({});
++ WithInteractions.play = async ({ canvasElement }) => {
++   const canvas = within(canvasElement);
++   // Simulates pinning the first task
++   await fireEvent.click(canvas.getByLabelText("pinTask-1"));
++   // Simulates pinning the third task
++   await fireEvent.click(canvas.getByLabelText("pinTask-3"));
++ };
+```
+
+Check your newly created story, and click the `Interactions` panel, and you'll see the following.
+
+<video autoPlay muted playsInline loop>
+  <source
+    src="/intro-to-storybook/storybook-interactive-stories-play-function.mp4"
+    type="video/mp4"
+  />
+</video>
+
+As you've seen, we're able to interact with our UI and quickly check how it responds if we update our tasks, ensuring it stays consistent. All without needing to spin up a testing environment or add additional packages.
```

Comment:
```suggestion
The play function allows us to interact with our UI and quickly check how it responds if we update our tasks. That keeps the UI consistent at no extra manual effort. All without needing to spin up a testing environment or add additional packages.
```

Reply:
Fixed with the latest commit for all frameworks.